docs/json.md

```diff
@@ -0,0 +1,423 @@
+
+## Lookering forward
+
+Current users of Looker SDKs may have noticed an `id` type may be either `integer` or `string`. For the stable (aka GA, generally available) release of API 4.0, all `id` references will be typed as `string`.
+Using strings for all ID references will allow Looker to scale out Looker services in the future, once ID generation does not require an auto-increment numeric ID value from a monolithic instance.
+
+We have worked to minimize the impact of this type change for all Looker-provided language SDKs. This document describes some JSON parsing requirements and how our SDKs support them.
+
+## JSON parsing requirements
+
+There are many requirements for parsing and producing JSON. Most requirements are identical among languages that support JSON **serialization** and **deserialization**. **Marshalling** and **unmarshalling** or **encoding** and **decoding** also refer to the same process.
+
+Serialization is the conversion of a JSON payload into a structure, record, or object of a given language. Deserialization is rendering an instantiated data type to its JSON representation.
+
+Almost all languages include standard first-party JSON serialization and deserialization support, though many have third-party offerings as the de facto standard.
+
+The parsing requirements of concern for the Looker language SDKs are:
+
+1. JSON value types conflicting with the API type specification. This is called [type ambiguity](#type-ambiguity) below. For the Looker API, these conflicts should be automatically resolvable for our SDKs. The conflict is either:
+   1. a numeric value for a string property (forward compatibility issue)
+   2. a string value for an integer property (backward compatibility issue)
+2. New properties in a JSON payload. These are ignored by the SDKs. (See [TypeScript caveats](#typescript-caveats) for additional comments.)
+3. Removed properties in a JSON payload. Typically, the property will end up unassigned. Some SDKs may throw errors with missing required properties, but that client code should be changed anyway if the property no longer exists.
+4. Name changes, like `space` to `folder` and `homepage` to `board`. There is no reasonable automatic solution for this, so type or property renames will require changing existing SDK client code.
+
+### What is forward and backward compatibility?
+
+In this document, **forward compatible** means an older SDK can process payloads from a newer Looker API server.
+
+**Backward compatible** means a newer SDK can process payloads from an older Looker API server.
+
+Our goal with the Looker language SDKs is to provide both forward and backward compatibility.
+
+The current Looker API server is written in Ruby, which is one of the aforementioned "flexibly typed" languages. The Looker API server accepts ID value references as **either** `integer` or `string` and converts them internally to the required type.
```

Comment:
> The current Looker API server is written in Ruby

is this public knowledge/ok-to-share?

Reply:
Which part, that it's written in Ruby? I don't think that's a trade secret or anything, but you had similar feedback as Jeremy, so I could drop the bit about Ruby and just state the API server behavior.

Reply:
updated the text